knimin/lib/data_access.py

```diff
@@ -2469,88 +2482,83 @@ def _study_is_unique(self, qiita_study_id=None, title=None, skip_id=None):
         Raises
         ------
         ValueError
-            If neither value is given, or either value already exists
-        """
-        cols = ['qiita_study_id', 'title']
-        vals = [qiita_study_id, title]
-        tags = [x.capitalize().replace('_', ' ').replace(' id', ' ID')
-                for x in cols]
-        if all(val is None for val in vals):
-            raise ValueError('Either %s or %s must be specified.'
-                             % (tags[0], tags[1]))
+            If the study is a duplicate
+        """
         with TRN:
             sql = """SELECT study_id
                      FROM pm.study
-                     WHERE {} = %s
-                     AND study_id IS DISTINCT FROM %s"""
-            errs = []
-            err_msg = "%s %s conflicts with exisiting study %s."
-            for col, val, tag in zip(cols, vals, tags):
-                if val:
-                    TRN.add(sql.format(col), [val, skip_id])
-                    res = TRN.execute_fetchflatten()
-                    if res:
-                        errs.append(err_msg % (tag, repr(val), res[0]))
-            if errs:
-                raise ValueError('\n'.join(errs))
+                     WHERE (study_id = %s OR title = %s)"""
+            sql_args = [qiita_study_id, title]
+            if skip_id:
+                sql += " AND study_id != %s"
+                sql_args.append(skip_id)
 
-    def create_study(self, qiita_study_id=None, title=None, alias=None,
-                     notes=None):
+            TRN.add(sql, sql_args)
+            res = TRN.execute_fetchflatten()
+            if res:
+                raise ValueError("Study (%s, %s) conflicts with studies %s"
```

Comment:
Tested here: https://github.com/biocore/labadmin/pull/187/files#diff-0d1d77e117ce5a979751aa7fba90e5e5R377

Note that is `fetchflatten` so res is a list of ints.